src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -394,8 +394,24 @@ private MethodIL Get_CORINFO_METHOD_INFO(MethodDesc method, MethodIL methodIL, C
             methodInfo->maxStack = (uint)methodIL.MaxStack;
             methodInfo->EHcount = (uint)methodIL.GetExceptionRegions().Length;
             methodInfo->options = methodIL.IsInitLocals ? CorInfoOptions.CORINFO_OPT_INIT_LOCALS : (CorInfoOptions)0;
-            methodInfo->regionKind = CorInfoRegionKind.CORINFO_REGION_NONE;
 
+            if (method.IsSharedByGenericInstantiations)
+            {
+                if (method.AcquiresInstMethodTableFromThis())
+                {
+                    methodInfo->options |= CorInfoOptions.CORINFO_GENERICS_CTXT_FROM_THIS;
+                }
+                else if (method.HasInstantiation)
```

Comment:
Could you make this go through the common helpers?

```csharp
if (method.AcquiresInstMethodTableFromThis())
{
    // ...
}
else if (method.RequiresInstMethodDescArg())
{
    // ...
}
else
{
    Debug.Assert(method.RequiresInstMethodTableArg());
    // ...
}
```

Reply:
(The outer `if` for `IsShared` won't be needed afterwards because the common helpers also check for that.)

Reply:
Good suggestion, thanks; fixed in 3rd commit.